pkgs/tools/security/pinentry/default.nix

```diff
@@ -1,40 +1,67 @@
 { fetchurl, fetchpatch, stdenv, lib, pkgconfig, autoreconfHook
 , libgpgerror, libassuan
-, libcap ? null, libsecret ? null, ncurses ? null, gtk2 ? null, gcr ? null
-, qt4 ? null, qt5 ? null
-, enableEmacs ? false
+, ncurses, gtk2, qt
+, libcap ? null, libsecret ? null, gcr ? null
+, enabledFlavours ? [ "curses" "tty" "gtk2" "qt" "gnome3" "emacs" ]
 }:
 
-assert qt5 != null -> qt4 == null;
-assert qt4 != null -> qt5 == null;
+with stdenv.lib;
+
+assert isList enabledFlavours && enabledFlavours != [];
 
```

Comment:
This does not seem to have been addressed. https://github.com/NixOS/nixpkgs/pull/49270/files#r228726960

Reply:
done. we can simply `assert inputsSatisfied`, because `mkEnablePinentry` is only called for every `enabledFlavor(s)`.